server/models/procedures/stock.sql

```diff
@@ -34,10 +34,137 @@ CREATE PROCEDURE ComputeMovementReference (
 )
 BEGIN
   DECLARE reference INT(11);
+  DECLARE flux INT(11);
 
   SET reference = (SELECT COUNT(DISTINCT document_uuid) AS total FROM stock_movement LIMIT 1);
+  SET flux = (SELECT flux_id FROM stock_movement WHERE document_uuid = documentUuid LIMIT 1);
 
   INSERT INTO `document_map` (uuid, text)
-  VALUES (documentUuid, CONCAT('MVT.', reference))
+  VALUES (documentUuid, CONCAT_WS('.', 'SM', flux, reference))
   ON DUPLICATE KEY UPDATE uuid = uuid;
 END $$
+
+-- post stock movement into vouchers
+DROP PROCEDURE IF EXISTS PostStockMovement;
```

Comment:
Very good stored procedure, it process once at all the voucher and voucher item inserting.